src/api/Api.cpp

```diff
@@ -4,33 +4,91 @@
 #include "config/Config.h"
 #include "maintenance/Logger.h"
 
-bool ApiClass::registerStation() {
+RegistrationResult* ApiClass::registerStation() {
+  RegistrationResult* result = new RegistrationResult();
+  result->ok = true;
+  if(isRegistered()) {
+    Logger::debug("[ApiClass::registerStation()] Station already registered");
+    result->message = "Station already registered";
+    return result;
+  }
+  
   accessToken = String("");
   accessTokenMillis = 0;
   if (Config::getRegistratonToken().equals("")) {
     Config::setRegistrationState(Config::RegistrationState::REGISTRATION_ERROR);
     Logger::debug("[ApiClass::registerStation()] Registration fail - no registration token");
-    return false;
+    result->message = "No registration token";
+    return result;
   }
 
   if (!Internet::isConnected()) {
     Config::setRegistrationState(Config::RegistrationState::REGISTRATION_ERROR);
     Logger::debug("[ApiClass::registerStation()] Registration fail - no internet connection");
-    return false;
+    result->message = "No internet connection";
+    return result;
+  }
+
+  Config::RegistrationState currentRegistrationState = Config::getRegistrationState();
+
+  if(currentRegistrationState < Config::RegistrationState::REGISTERED) {
+    if(!doRegister(result)) result;
```

Comment:
Shouldn't there be:
```suggestion
    if(!doRegister(result)) return result;
```
?
What's more, I was once told that inline *if*'s are bad, but it will take more space, tbh I don't know what option is better:
```suggestion
    if(!doRegister(result)) {
        return result;
    }
```